addons/website_sale/models/website.py

```diff
@@ -270,11 +270,11 @@ def get_current_pricelist(self):
             if available_pricelists and pricelist not in available_pricelists:
                 # If there is at least one pricelist in the available pricelists
                 # and the chosen pricelist is not within them
-                # it then choose the first available pricelist.
+                # it then choose the first available pricelist based on sequence.
                 # This can only happen when the pricelist is the public user pricelist and this pricelist is not in the available pricelist for this localization
                 # If the user is signed in, and has a special pricelist (different than the public user pricelist),
                 # then this special pricelist is amongs these available pricelists, and therefore it won't fall in this case.
-                pricelist = available_pricelists[0]
+                pricelist = available_pricelists.sorted('sequence')[0]
```

Comment:
```suggestion
                pricelist = available_pricelists.sorted()[0]
```
I believe it'd be better to follow the full `_order` from the model, and not only the sequence.

Reply:
Took a look to be sure before telling you :D
Not the best perf, since it makes another request to the database, but I believe it's still better to follow the generic specification as much as possible (and work if custom modules modify the `_order` too).